src/main/java/com/conveyal/datatools/manager/controllers/api/GtfsPlusController.java

```diff
@@ -252,20 +250,36 @@ private static String publishGtfsPlusFile(Request req, Response res) {
     /**
      * HTTP endpoint that validates GTFS+ tables for a specific feed version (or its saved/edited GTFS+).
      */
-    private static Collection<ValidationIssue> getGtfsPlusValidation(Request req, Response res) {
+    private static GtfsPlusValidation getGtfsPlusValidation(Request req, Response res) {
         String feedVersionId = req.params("versionid");
-        List<ValidationIssue> issues = new LinkedList<>();
+        GtfsPlusValidation gtfsPlusValidation = null;
         try {
-            issues = validateGtfsPlus(feedVersionId);
+            gtfsPlusValidation = GtfsPlusValidation.validate(feedVersionId);
         } catch(IOException e) {
             logMessageAndHalt(req, 500, "Could not read GTFS+ zip file", e);
         }
-        return issues;
+        return gtfsPlusValidation;
+    }
+
+    /**
+     * HTTP endpoint to delete the GTFS+ specific edits made for a feed version. In other words, this will revert to
+     * referencing the original GTFS+ files for a feed version. Note: this will not delete the feed version itself.
+     */
+    private static String deleteGtfsPlusFile(Request req, Response res) {
+        String feedVersionId = req.params("versionid");
+        File file = gtfsPlusStore.getFeed(feedVersionId);
+        if (file == null) {
+            logMessageAndHalt(req, HttpStatus.NOT_FOUND_404, "No GTFS+ file found for feed version");
+            return null;
+        }
+        file.delete();
+        return SparkUtils.formatJSON("message", "GTFS+ edits deleted successfully.");
     }
 
     public static void register(String apiPrefix) {
         post(apiPrefix + "secure/gtfsplus/:versionid", GtfsPlusController::uploadGtfsPlusFile, JsonUtil.objectMapper::writeValueAsString);
         get(apiPrefix + "secure/gtfsplus/:versionid", GtfsPlusController::getGtfsPlusFile);
+        delete(apiPrefix + "secure/gtfsplus/:versionid", GtfsPlusController::deleteGtfsPlusFile);
         get(apiPrefix + "secure/gtfsplus/:versionid/timestamp", GtfsPlusController::getGtfsPlusFileTimestamp, JsonUtil.objectMapper::writeValueAsString);
         get(apiPrefix + "secure/gtfsplus/:versionid/validation", GtfsPlusController::getGtfsPlusValidation, JsonUtil.objectMapper::writeValueAsString);
         post(apiPrefix + "secure/gtfsplus/:versionid/publish", GtfsPlusController::publishGtfsPlusFile, JsonUtil.objectMapper::writeValueAsString);
```

Comment:
Should this publish endpoint also create a new entry in the GTFS_PLUS_SUBDIR? I tried publishing a new GTFS+ in the UI and it hit this endpoint, but I don't see how it's supposed to result in creating a GTFSPlusValidation that has the `published` field set to true.

Reply:
In this context, `published` is essentially a proxy for `unedited`. Once the new feed is published, it becomes a new feed version, which moves it out of the GTFS+ work flow. The edits are preserved in place so that they can be reverted at any time or new edits can be added.

Reply:
In this context, `published` is essentially a proxy for `unedited`. Once the new feed is published, it becomes a new feed version, which moves it out of the GTFS+ work flow. The edits are preserved in place so that they can be reverted at any time or new edits can be added.